File: N3.15/N3.15.cpp
#include <iostream>

using namespace std;

bool ReglamentCheck(int num)
{
    int PrevDigit = num % 10;
    num = num / 10;
    while (num != 0)
    {
        int CurrentDigit = num % 10;
        if (CurrentDigit <= PrevDigit) // Изменено условие
        {
            PrevDigit = CurrentDigit;
            num /= 10;
        }
        else
        {
            return false;
        }
    }
    return true;
}

int NumReverser(int numblank)
{
    if (ReglamentCheck(numblank))
    {
        int ReversedNum = 0;
        while (numblank != 0)
        {
            ReversedNum = ReversedNum * 10 + numblank % 10;
            numblank /= 10;
        }
        return ReversedNum;
    }
    else
    {
        return 0;
    }
}

int main()
{
    int n;
    cout << "Enter the number of numbers : ";
    cin >> n;
    
    if (n > 0)
    {
        cout << "Enter " << n << " numbers:" << endl;
        
        for (int i = 0; i < n; ++i)
        {
            int number;
            cin >> number;
            cout << NumReverser(number);
            if (i < n - 1) // Проверяем, нужно ли выводить пробел
            {
                cout << " ";
            }
        }
        cout << endl;
    }

    return 0;
}

File: N3.15/N3.15.js
function reglamentCheck(num) {
    let prevDigit = num % 10;
    num = Math.floor(num / 10);
    while (num !== 0) {
      let currentDigit = num % 10;
      if (currentDigit <= prevDigit) {
        prevDigit = currentDigit;
        num = Math.floor(num / 10);
      } else {
        return false;
      }
    }
    return true;
  }
  
  function numReverser(numBlank) {
    if (reglamentCheck(numBlank)) {
      let reversedNum = 0;
      while (numBlank !== 0) {
        reversedNum = reversedNum * 10 + numBlank % 10;
        numBlank = Math.floor(numBlank / 10);
      }
      return reversedNum;
    } else {
      return 0;
    }
  }
  
  const defaultCount = 4;
  let n = defaultCount;
  
  for (let i = 0; i < n; ++i) {
    let number = parseInt(prompt("Enter a number:"));
    console.log(numReverser(number) + " ");
  }
  console.log();